File: Python/ZigZag Conversion.py
class Solution:
    # @return a string
    def convert(self, s, nRows):
        if len(s) == 1 or nRows == 1:
            return s
        seq = []
        for i in range(nRows):
            seq.append([])
        cnt = 0
        index = 0
        while index < len(s):
            while index < len(s) and cnt + 1 < nRows:
                seq[cnt].append(s[index])
                index += 1
                cnt += 1
            while index < len(s) and cnt - 1 >= 0:
                seq[cnt].append(s[index])
                index += 1
                cnt -= 1
        st = ""
        for i in range(len(seq)):
            for j in range(len(seq[i])):
                st += seq[i][j]
        return st

File: C++/ZigZag-Conversion.cpp
class Solution {
public:
    string convert(string s, int numRows) {
        if (numRows == 1) return s;
        vector<vector<char> > res(numRows);
        int cur = 0, index = 0, len = s.size();
        while (index < len) {
            while (cur + 1 < numRows && index < len) {
                res[cur ++].push_back(s[index ++]);
            }
            while (cur > 0 && index < len) {
                res[cur --].push_back(s[index ++]);
            }
        }
        string str;
        for (int i = 0; i < numRows; i ++) {
            for (int j = 0; j < res[i].size(); j ++) {
                str += res[i][j];
            }
        }
        return str;
    }
};